internal/server/service_reinitialize_roots.go

```diff
@@ -0,0 +1,38 @@
+package server
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/nodeenrollment"
+	"github.com/hashicorp/nodeenrollment/types"
+)
+
+// ReinitializeRoots is a domain service function that removes both root certificates and then
+// calls RotateRoots to generate new root certificates.
+// Accepts the nodeenrollment option, WithCertificateLifetime(time.Duration) to specify the lifetime
+// of the generated cert(s)
+func ReinitializeRoots(ctx context.Context, workerAuthRepo *WorkerAuthRepositoryStorage, opt ...nodeenrollment.Option) error {
+	const op = "server.ReinitializeRoots"
+	if workerAuthRepo == nil {
+		return errors.New(ctx, errors.InvalidParameter, op, "missing workerAuthRepo")
+	}
+
+	// Delete roots from workerAuthRepo
+	rootCerts := &types.RootCertificates{Id: CA_ID}
+	err := workerAuthRepo.Load(ctx, rootCerts)
+	if err != nil {
+		return errors.Wrap(ctx, err, op)
+	}
+	err = workerAuthRepo.Remove(ctx, rootCerts)
```

Comment:
I think we should fold this functionality into an option in RotateRoots (and from there an option into nodeenrollment). By doing so if our storage is transactional, the removal and rotation happens in the same transaction. It'd be simple to implement in nodeenrollment, it just calls Remove on the existing values at the start of the function if the option is set.

Reply:
In fact if you do this I think you can get rid of this function entirely in favor of calling RotateRoots with an option indicating to clear storage beforehand.

Reply:
I'll update nodeenrollment to support the new option

Reply:
https://github.com/hashicorp/nodeenrollment/pull/28